src/main/java/io/github/thebusybiscuit/exoticgarden/PlantsListener.java

```diff
@@ -50,12 +51,72 @@ public PlantsListener(ExoticGarden plugin) {
 
     @EventHandler
     public void onGrow(StructureGrowEvent e) {
+        if (PaperLib.isPaper()) {
+            if (PaperLib.isChunkGenerated(e.getLocation())) {
+                onGrowSf(e);
+            } else {
+                PaperLib.getChunkAtAsync(e.getLocation()).thenRun(() -> onGrowSf(e));
+            }
+        } else {
+            if (!e.getLocation().getChunk().isLoaded()) {
+                e.getLocation().getChunk().load();
+            }
+            onGrowSf(e);
+        }
+    }
+
+    @EventHandler
+    public void onGenerate(ChunkPopulateEvent e) {
+        if (!SlimefunPlugin.getWorldSettingsService().isWorldEnabled(e.getWorld())) {
+            return;
+        }
+
+        if (!cfg.getStringList("world-blacklist").contains(e.getWorld().getName())) {
+            Random random = ThreadLocalRandom.current();
+            if (random.nextInt(100) < cfg.getInt("chances.BUSH")) {
+                Berry berry = ExoticGarden.getBerries().get(random.nextInt(ExoticGarden.getBerries().size()));
+                if (berry.getType().equals(PlantType.ORE_PLANT)) return;
+
+                int x = (e.getChunk().getX() * 16 + random.nextInt(16));
+                int z = e.getChunk().getZ() * 16 + random.nextInt(16);
+
+                if ((x < 29999983 && x > -29999983) && (z < 29999983 && z > -29999983)) {
```

Comment:
Where is that magic number coming from?
I am guessing it's the world limit or such... But since you reuse this number a total of 8 times in the code you should probably make it a `private static final` constant in this class and give it a name that explains its meaning.

Are these checks here necessary in the first place?

Reply:
It's necessary with PaperLib, if you try to load a chunk over the limit it gives an error.
https://paste.gg/p/anonymous/d410687620cc4517a6e43358e00e5293

Reply:
Thanks, good to know :). I'd still recommend turning this into a static constant though rather than having this "magic number" appear 8 times.